src/XRP/model/xrp-signer.ts

```diff
@@ -13,22 +16,50 @@ export default class XrpSigner {
    * @return an XrpSigner with its fields set via the analogous protobuf fields.
    * @see https://github.com/ripple/rippled/blob/develop/src/ripple/proto/org/xrpl/rpc/v1/transaction.proto#L90
    */
-  public static from(signer: Signer): XrpSigner | undefined {
+  public static from(signer: Signer, xrplNetwork: XrplNetwork): XrpSigner {
     const account = signer.getAccount()?.getValue()?.getAddress()
+    if (!account) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Signer protobuf is missing `account` field.',
+      )
+    }
+    const accountXAddress = XrpUtils.encodeXAddress(
+      account,
+      undefined,
+      xrplNetwork == XrplNetwork.Test || xrplNetwork == XrplNetwork.Dev,
+    )
     const signingPublicKey = signer.getSigningPublicKey()?.getValue_asU8()
+    if (!signingPublicKey) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Signer protobuf is missing `SigningPublicKey` field.',
+      )
+    }
     const transactionSignature = signer
       .getTransactionSignature()
       ?.getValue_asU8()
-    return new XrpSigner(account, signingPublicKey, transactionSignature)
+    if (!transactionSignature) {
+      throw new XrpError(
+        XrpErrorType.MalformedProtobuf,
+        'Signer protobuf is missing `TnxSignature` field.',
```

Comment:
```suggestion
        'Signer protobuf is missing `transactionSignature` field.',
```

Reply:
i guess we decided long ago in SDK land that we don't abbreviate "transaction".  Unless this is actually how the proto is written?? in which case go for it, though I think it would maybe be spelled "txn" instead of "tnx"

Reply:
It's how it was written in the docs. The `Tnx` was probably a typo though.